packages/adapters-library/src/adapters/snapshots.test.ts

```diff
@@ -233,6 +234,141 @@ function runProductTests(
           )
         })
       })
+
+      // Only run user event tests if there are positions test cases with txHash defined and userEvent configured in the adapter
+      const hasUserEventTest = positionsTestCases.some(
+        (testCase) =>
+          testCase.input.openingPositionTxHash &&
+          !!defiProvider.adaptersController.fetchAdapter(
+            testCase.chainId,
+            protocolId,
+            productId,
+          ).adapterSettings.userEvent,
+      )
+
+      if (hasUserEventTest) {
+        describe('userEventExistsInTransaction', () => {
+          it.each(
+            positionsTestCases.map((testCase) => [testKey(testCase), testCase]),
+          )(
+            'User event for test %s match',
+            async (_, testCase) => {
+              const { defiProvider } = await fetchSnapshot(
+                testCase,
+                protocolId,
+                productId,
+              )
```

Comment:
nit, but we can use the checks to exit early instead of creating multiple levels of indentation?

Something like:

```
const adapter = defiProvider.adaptersController.fetchAdapter(
  testCase.chainId,
  protocolId,
  productId,
)

if (!testCase.input.openingPositionTxHash || !adapter?.adapterSettings.userEvent) {
  return
}
```

Happy to merge it anyway, we can polish this things later and this is not runtime code.